### DISCUSSION TITLE
No isce.log file

### DISCUSSION THREAD

**Discussion by shubhayanroychowdhury** [alexisInSAR/EZ-InSAR#59] [Help]
@insarwxw @alexisInSAR 
Hi,
When I run topsApp.py -h for testing purpose, I got permission error.
![Screenshot from 2023-08-21 13-57-01](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/9743b00a-7786-461b-bf20-397cbdf7c818)

When I go to applications folder, there is no isce.log file.
![Screenshot from 2023-08-21 13-59-41](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/503043e2-bbca-499c-9835-9d80344f0dd7)
Thank you for your time and consideration.

Best regards, 
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
Hi, 

I assume that this error is not abnormal because you have run the script in the "script/software" directory. Please run the same script in your working directory: the log file should be created with the correct permissions. 

Kind regards, 

Alexis

**Comment by shubhayanroychowdhury** [comment]
@alexisInSAR 
Hi,

I apologize if I've misunderstood anything, but I was wondering if the "topsApp.py" file is located within the application folder, Would you kindly clarify whether I should run this script in the application directory or if there is a different directory where it needs to be executed? Your assistance would be greatly appreciated.

Thank you for your time.

best regards,
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
If your PATH variable is correct (via .bashrc or config file), you can run the ISCE scripts anywhere. 

Kind regards, 

Alexis

**Comment by shubhayanroychowdhury** [comment]
@alexisInSAR 
Thank you so much for your guidance. And one more thing I would like to ask, I got this error then when I run 'python PyAPS/tests/test_dload.py'.
![Screenshot from 2023-08-21 15-36-21](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/0358ea71-0ff4-4148-8a31-8052f1b2e275)

Best regards, 
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
I think you have forgotten to run the correct conda environment. 
````
conda activate InSARenv
````

Try again. If the command does not work: 
````
conda install -c conda-forge pyaps3
````

Kind regards, 

Alexis

**Comment by shubhayanroychowdhury** [comment]
I run commands as you mentioned, now I got this error.
![Screenshot from 2023-08-21 15-44-33](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/d34343be-3695-4f3d-a2e3-15af75d24fbd)
Thank you for giving me your time.

Best regards,
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
Okay. The command works for me. The error is not linked to the Python environment (missed files for example). 

Kind regards, 

Alexis

**Comment by shubhayanroychowdhury** [comment]
Can you give me any suggestion on how to solve this error?. That would be really helpful for me.

Best regards,
Shubhayan Roy Chowdhury

**Comment by shubhayanroychowdhury** [comment]
@alexisInSAR 
hi,
Also, I did not find this mt_prep_isce. 
![Screenshot from 2023-08-21 17-08-14](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/94e03a0b-1e40-4222-9728-66954793031c)
I only get prep_isce in Stamps folder that is
![Screenshot from 2023-08-21 17-10-57](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/d6691141-7670-433a-9287-bbcbf89c4453)
I am actually new, It really helps me if you guide me.

 best regards
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
Normal, this function is in the **_Contrib_** directory with ISCE. 

Alexis

**Comment by shubhayanroychowdhury** [comment]
@alexisInSAR 
Hi,
I did not find this mt_prep_isce in contrib directory. 

![Screenshot from 2023-08-22 16-40-40](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/8eb632da-07af-4766-9122-9a6ada542a31)

Best regards,
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
What is the output of: 
````
echo $PATH 
````

**Comment by shubhayanroychowdhury** [comment]
Hi,
This is the output of
![Screenshot from 2023-08-23 18-55-01](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/e6df7026-3381-4caf-a5c8-db119b08ef05)


Best regards,
Shubhayan Roy Chowdhury

**Comment by alexisInSAR** [comment]
Okay. 

For me, all the required directories are in this variable. However, I don't understand why you have a /home2/ directory? 

Where is ISCE installed? What is its path?

**Comment by shubhayanroychowdhury** [comment]
Hi,
Should I keep all files into one directory?
This is the path.
![Screenshot from 2023-08-24 11-05-51](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/da3c8f4a-d0f8-40af-85a9-e763e17a660a)

Best regards

**Comment by alexisInSAR** [comment]
Okay. You have two solutions: 
1. you can add this path in the PATH variable: **_/home2/tools_insar/proc_insar/tool_DIR/ISCE/contrib/timeseries/prepStackToStaMPS/bin_**
2. The installation directories are not correct. _tool_DIR_ should not be a directory because it should be a bash variable. I assume that a mistake was made during the installation. 

Kind regards, 

Alexis

**Comment by shubhayanroychowdhury** [comment]
Hi,

Thank you so much for your time and guidance. I will do it and let you know.

Best regards

**Comment by shubhayanroychowdhury** [comment]
@alexisInSAR  @insarwxw 
hi,
![Screenshot from 2023-09-07 15-19-44](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/ea1a5eaa-b689-498c-8fe8-8e6fdfa626ef)

When I click **check the SLCs**, I got this error.

![Screenshot from 2023-09-07 15-20-03](https://github.com/alexisInSAR/EZ-InSAR/assets/129718401/4bb8773d-a809-4437-9f0a-b53d68a8b244)

How do i solve this error?

best regards,
Shubhayan Roy Chowdhury
